python/paddle/fluid/dygraph/io.py

```diff
@@ -645,6 +651,329 @@ def _construct_params_and_buffers(model_path,
+    # will be SelectedRows, not LoDTensor. But tracer will just
+    # set param grad VarBase by forward VarBase(LoDTensor)
+    # If we don't change grad_var type here, RunProgramOp need
+    # transform SelectedRows to LoDTensor forcibly, it may not
+    # be user wanted result.
+    for persistable_var in persistable_vars:
+        grad_var_name = var.name + core.grad_var_suffix()
+        grad_var = trace_program.block(0).find_var(cpt.to_bytes(grad_var_name))
+        # NOTE: cannot find var desc maybe not problem, 
+        # such as in batch_norm
+        if grad_var is None:
+            continue
+        persistable_var._set_grad_type(grad_var.type())
+
+    # 3. prepare output, keep same form with inputs
+    outs = output_vars
+    if len(output_vars) == 1:
+        outs = output_vars[0]
+    return outs
+
+
+def _run_static_graph(input, program_holder, trace_program):
+    main_program = framework.default_main_program()
+    param_var_names = _get_persistable_var_names(trace_program)
+    dict_rename_var_old_new = None
+    _, dict_rename_var_old_new = _rename_var_program_desc(
+        trace_program, exclude=param_var_names)
+    trace_program.flush()
+    output_names = [var.name() for var in program_holder.output_descs]
+    # append blocks from 'trace_program'
+    _append_block(main_program, trace_program, program_holder, input,
+                  dict_rename_var_old_new)
+    main_program._sync_with_cpp()
+    outs = _get_output_from_program(main_program, program_holder,
+                                    dict_rename_var_old_new)
+    if len(outs) == 1:
+        outs = outs[0]
+    return outs
+
+
+def _collect_parent_var(program, block_idx):
+    '''
+    Get variables in current block and its parent block.
+    
+    Args:
+        program(Program): The program containing the current block.
+        block_idx(int): index of current block.
+
+    Returns:
+        List: list of variable.
+    '''
+    vars = []
+    if block_idx < 0:
+        return vars
+    for var in program.block(block_idx).vars:
+        vars.append(var)
+    parent_idx = program.block(block_idx).parent_idx
+    if parent_idx > -1:
+        vars += _collect_parent_var(program, parent_idx)
+    return vars
+
+
+def _append_block(dest_program,
+                  src_program_desc,
+                  program_holder,
+                  input_names,
+                  dict_rename_var_old_new=None):
+    '''
+    Append Variables and Operators in 'src_program_desc' to dest_program.
+    
+    Args:
+        dest_program(Program): Variables and Operators are appended to it.
+        src_program_desc(ProgramDesc): index of current block.
+        program_holder(_ProgramHolder): program_holder of TranslatedLayer
+        input_names(list): list of input variables
+        dict_rename_var_old_new(None|dict): When using '_rename_var_program_desc', 
+        use it to map the variables in 'program_holder' and 'program_holder'.
```

Comment:
Is this a comment error?

mapping the variables in ''program_holder' and 'program_holder', two same things?

Reply:
Done,thx.